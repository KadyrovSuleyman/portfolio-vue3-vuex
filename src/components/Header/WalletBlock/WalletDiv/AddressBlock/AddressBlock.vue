<script setup lang="ts">

import { computed } from 'vue';
import Div from '@/elements/Div/Div.vue';
import Span from '@/elements/Span/Span.vue';
import Button from '@/elements/Button/Button.vue';
import propsObj from '@/elements/propsObj';

type OnClickHandlerT = (payload: MouseEvent) => void;
// eslint-disable-next-line no-undef
const props = defineProps({
  ...propsObj,

  onClick: {
    type: Function,
    default: (): OnClickHandlerT => () => ({}),
  },
});

const comp = computed(() => ({ elem: props.elem || 'addressBlock' }));

</script>

<template>
  <Div :block="props.block" :elem="comp.elem" :mods="props.mods">
    <Span :block="comp.elem">
      <slot></slot>
    </Span>
    <Button :block="comp.elem" :icon="'copy'" @click="onClick"/>
  </Div>
</template>

<style lang="scss">
  @use 'styles' as *;
</style>
