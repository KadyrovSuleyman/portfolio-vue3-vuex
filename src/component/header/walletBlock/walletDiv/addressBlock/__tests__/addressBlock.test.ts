import { mount, VueWrapper } from '@vue/test-utils';
import { createStore } from 'vuex';
import AddressBlock from '../addressBlock.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('connectBlock renders', () => {
  wrapper = mount(AddressBlock);
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(AddressBlock);
  expect(wrapper.find('.addressBlock').classes()).toEqual(['addressBlock']);

  await wrapper.setProps({
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.addressBlock').classes()).toEqual(['addressBlock', 'addressBlock__selected']);

  await wrapper.setProps({
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.addressBlock').classes()).toEqual(['addressBlock']);
});

it('click sends data to outer store', async () => {
  const store = createStore({
    state: {
      clicked: false,
    },
    mutations: {
      click: (state) => { state.clicked = true; },
    },
  });

  wrapper = mount(AddressBlock, {
    global: {
      plugins: [store],
    },
    props: {
      onClick: store.commit.bind('click'),
    },
  });

  expect(store.state.clicked).toBeFalsy();

  await wrapper.find('button').trigger('click');
  expect(store.state.clicked).toBeTruthy();
});

it('watchs changes from outer store', async () => {
  const wr = mount({
    data() {
      return {
        text: 'before',
      };
    },
    methods: {
      change() {
        this.text = 'after';
      },
    },
    components: {
      AddressBlock,
    },
    template: `
      <div>
        <AddressBlock>{{ text }}</AddressBlock>
        <button class="test-btn" @click="change"></button>
      </div>
    `,
  });
  expect(wr.find('span').text()).toBe('before');

  await wr.find('.test-btn').trigger('click');
  expect(wr.find('span').text()).toBe('after');

  wr.unmount();
});
