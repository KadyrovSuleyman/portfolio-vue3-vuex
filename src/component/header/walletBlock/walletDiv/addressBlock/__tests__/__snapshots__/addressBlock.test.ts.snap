// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectBlock renders 1`] = `<div class="addressBlock"><span class="addressBlock-span"></span><button class="addressBlock-button addressBlock-button__icon_copy"></button></div>`;
